### PR TITLE
bugfix: insecure flag not working for OCI registries with auth

### PR DESCRIPTION
**Pull request by juan131**
## Description

Dealing with OCI registries was failing when authentication was required if the registry used invalid certificates **even if the insecure flag was provided**. Check the error below:

```console
    ⚠️  Failed syncing "aaa-x.y.z" chart: Failed to push Helm chart: failed to authorize: failed to fetch oauth token: Post "https://REGISTRY_HOST/service/token": tls: failed to verify certificate: x509: “REGISTRY_HOST” certificate is not trusted
    ✘  Error syncing charts: Failed to push Helm chart: failed to authorize: failed to fetch oauth token: Post "https://REGISTRY_HOST/service/token": tls: failed to verify certificate: x509: “REGISTRY_HOST” certificate is not trusted
```

## Changes Made

This PR addresses this issue by ensuring we use an appropriate TLS client configuration when creating the docker authorizer

### CODE REVIEW

File: pkg/artifacts/helm.go [hunk 1]
```diff
@@ -181,21 +189,19 @@ func PullChart(chartURL, version string, destDir string, opts ...RegistryClientO
...
-	cfg.RegistryClient = reg.client
-	client := action.NewPushWithOpts(action.WithPushConfig(cfg))
 
+	client := action.NewPushWithOpts(
+		action.WithPushConfig(&action.Configuration{RegistryClient: reg.client}),
+	)
 	client.Settings = cli.New()
 
-	if _, err := client.Run(tarFile, pushChartURL); err != nil {
-		return fmt.Errorf("failed to push Helm chart: %w", err)
```
Comment (mpermar) [inline pkg/artifacts/helm.go:195]: It looks like you swallowed this error message with your change, i.e. if we merge this PR we won't be showing the user an actual human friendly message.

File: pkg/artifacts/helm.go [hunk 2]
```diff
@@ -80,38 +80,44 @@ func NewRegistryClientConfig(opts ...RegistryClientOption) *RegistryClientConfig
...
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: cfg.UseInsecureHTTPS, // #nosec G402
+			},
+		},
+	}
+
 	if cfg.UsePlainHTTP {
 		opts = append(opts, registry.ClientOptPlainHTTP())
 	} else {
-		if cfg.UseInsecureHTTPS { // #nosec G402
```
Comment (mpermar) [inline pkg/artifacts/helm.go:88]: After your change, do we then need this cfg option anymore? If not, then I think it needs to be cleaned up from the codebase.

**Review by mpermar — CHANGES_REQUESTED** [review]

File: pkg/artifacts/helm.go (hunk 1 above)
Comment (juan131) [reply pkg/artifacts/helm.go:195]: It stuttered before this change, see this example:
```
Failed to push Helm chart: failed to push Helm chart: failed to do request: Head "https://harbor.juan.ariza/v2/charts/grafana/blobs/sha256:ff28b728fcd253503bd27c8dbb0a765d4f89a2b41d88c5991219d8cfec1389b6": tls: failed to verify certificate: x509: “harbor.juan.ariza” certificate is not trusted
```

File: pkg/artifacts/helm.go (hunk 2 above)
Comment (juan131) [reply pkg/artifacts/helm.go:88]: Yes, we need it. It's used here:
```golang
	httpClient := &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				InsecureSkipVerify: cfg.UseInsecureHTTPS, // #nosec G402
			},
		},
	}
````

**Review by mpermar — APPROVED** [review]

**Pull request merged**